### PR TITLE
Weekly Patch Release v.1.2.5 [full merge, no squash]

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Refers to #6083

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
still missing metrics and #6506 needs to separate fix from API chnage

**Comment by Borda** [comment]
the same error as in https://github.com/PyTorchLightning/pytorch-lightning/pull/6536#issuecomment-802942487

**Review by awaelchli — APPROVED** [review]
🎉

**Review by kaushikb11 — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Comment by Borda** [comment]
missing #6536 as the test does not pass, probably needs some other commit
and #6506 as there mixed API changes so @SeanNaren will separate the fix 🐰

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/metrics/test_metric_lightning.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import torch
+from torchmetrics import Metric as TMetric
 
 from pytorch_lightning import Trainer
-from pytorch_lightning.metrics import Metric, MetricCollection
+from pytorch_lightning.metrics import Metric as PLMetric
+from pytorch_lightning.metrics import MetricCollection
```
Comment (Borda) [inline tests/metrics/test_metric_lightning.py:6]: @tchaton van you pls double check that this is sufficient compatibility test for PL and TM metrics

File: CHANGELOG.md [hunk 2]
```diff
@@ -139,9 +42,6 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed when Train loop config was run during `Trainer.predict` ([#6541](https://github.com/PyTorchLightning/pytorch-lightning/pull/6541))
```
Comment (carmocca) [inline CHANGELOG.md:42]: what happened to the 1.2.4 changelog? can you fix it in this PR?

Comment (Borda) [reply CHANGELOG.md:42]: what do you mean, the three entries below were wrongly added before and some were duplicate, if you check the tree they are missing in v1.2.x
